src/txn/txn.c

```diff
@@ -1009,9 +1009,16 @@ __txn_fixup_prepared_update(
      * Transaction error and prepare are cleared temporarily as cursor functions are not allowed
      * after an error or a prepared transaction.
      */
-    txn_flags = FLD_MASK(txn->flags, WT_TXN_ERROR | WT_TXN_PREPARE);
+    txn_flags = FLD_MASK(txn->flags, WT_TXN_ERROR);
     F_CLR(txn, txn_flags);
 
+    /*
+     * The WT_TXN_PREPARE flag should not be cleared as it may cause issues where a prepared
+     * transaction gets forcibly rolled back. A new flag is set instead, to allow cursor functions
+     * after a prepared transaction.
+     */
+    F_SET(session, WT_SESSION_PREPARE_TXN_IN_PROGRESS);
```

Comment:
I'd rework this comment to explain why we have to do anything at all: "The API layer will immediately return an error if XXX; we can't clear the txn-prepare flag because YYY, so we have this new flag ZZZ."

Reply:
I didn't have a strong opinion about this, but thought that because there is already a prepare flag in the txn structure, it may be confusing if we have two with similar names. I'm happy to change the location though.

Reply:
Maybe `WT_TXN_PREPARE_IGNORE_API_CHECK`. :)